x-pack/plugins/security_solution/server/endpoint/services/actions/action_details_by_id.test.ts

```diff
@@ -0,0 +1,166 @@
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { ElasticsearchClientMock } from '@kbn/core/server/mocks';
+import { elasticsearchServiceMock } from '@kbn/core/server/mocks';
+import * as estypes from '@elastic/elasticsearch/lib/api/typesWithBodyKey';
+import {
+  EndpointAction,
+  EndpointActionResponse,
+  LogsEndpointAction,
+  LogsEndpointActionResponse,
+} from '../../../../common/endpoint/types';
+import { EndpointActionGenerator } from '../../../../common/endpoint/data_generators/endpoint_action_generator';
+import { getActionDetailsById } from '..';
+import { NotFoundError } from '../../errors';
+import {
+  applyActionsEsSearchMock,
+  createActionRequestsEsSearchResultsMock,
+  createActionResponsesEsSearchResultsMock,
+} from './mocks';
+
+describe('When using `getActionDetailsById()', () => {
+  let esClient: ElasticsearchClientMock;
+  let endpointActionGenerator: EndpointActionGenerator;
+  let actionRequests: estypes.SearchResponse<EndpointAction | LogsEndpointAction>;
+  let actionResponses: estypes.SearchResponse<EndpointActionResponse | LogsEndpointActionResponse>;
+
+  beforeEach(() => {
+    esClient = elasticsearchServiceMock.createScopedClusterClient().asInternalUser;
+    endpointActionGenerator = new EndpointActionGenerator('seed');
+
+    actionRequests = createActionRequestsEsSearchResultsMock();
+    actionResponses = createActionResponsesEsSearchResultsMock();
+
+    applyActionsEsSearchMock(esClient, actionRequests, actionResponses);
+  });
+
+  it('should return expected output', async () => {
+    await expect(getActionDetailsById(esClient, '123')).resolves.toEqual({
+      agents: ['agent-a'],
+      command: 'isolate',
+      completedAt: '2022-04-30T16:08:47.449Z',
+      id: '123',
+      isCompleted: true,
+      isExpired: false,
+      logEntries: [
+        {
+          item: {
+            data: {
+              '@timestamp': '2022-04-27T16:08:47.449Z',
+              action_id: '123',
+              agents: ['agent-a'],
+              data: {
+                command: 'isolate',
+                comment: '5wb6pu6kh2xix5i',
+              },
+              expiration: '2022-04-29T16:08:47.449Z',
+              input_type: 'endpoint',
+              type: 'INPUT_ACTION',
+              user_id: 'elastic',
+            },
+            id: '44d8b915-c69c-4c48-8c86-b57d0bd631d0',
+          },
+          type: 'fleetAction',
+        },
+        {
+          item: {
+            data: {
+              '@timestamp': '2022-04-30T16:08:47.449Z',
+              action_data: {
+                command: 'unisolate',
+                comment: '',
+              },
+              action_id: '123',
+              agent_id: 'agent-a',
+              completed_at: '2022-04-30T16:08:47.449Z',
+              error: '',
```

Comment:
Assuming this is a completed successful response the `error` key shouldn't be here